File: csharp/src/Behavioral/State/DvdStateNameStars.cs
namespace Behavioral.State {


//DvdStateNameStars.java - two of two Concrete States

public class DvdStateNameStars : DvdStateName {  
    int starCount;
    
    public DvdStateNameStars() {
        starCount = 0;
    }
    
    public string showName(DvdStateContext dvdStateContext, 
                         string nameIn) {
         string retval = nameIn.Replace(' ','*');
         // show stars twice, switch to exclamation point
         if (++starCount > 1) {
             dvdStateContext.setDvdStateName( new DvdStateNameExclaim());
         }       
			return retval;
    }  
}

}

File: go/behavioral/state/DvdStateNameStars.go


//DvdStateNameStars - two of two Concrete States

package state

import (
	"strings"
)

type DvdStateNameStars struct { // implements DvdStateName
	starCount int
}

func (this *DvdStateNameStars) ShowName ( ctx *DvdStateContext , nameIn string ) string  {
	this.starCount++
	retval := strings.Replace(nameIn, " ","*",this.starCount )
	if this.starCount > 1 {
		ctx.Name = new(DvdStateNameExclaim)
	}
	return retval
}
